### PR TITLE
Api26/jobschedule reader update service

### PR DESCRIPTION
**Pull request by mzorz**
This PR implements a distinction on pre and post API 26 and makes use of the standard Service or a `JobService` starting from API 26, for the `ReaderUpdateService` as explained in https://github.com/wordpress-mobile/WordPress-Android/pull/7506/#issuecomment-378949294.

In order to do that, the following steps were made:
- extracted the business logic from `ReaderUpdateService` out to a shared class named `ReaderUpdateLogic`.
- Made a specific implementation of the `JobService` for API 26+, thus effectively having differentiated Services
- Declared such a new JobService in the manifest
- a new interface, called `ServiceCompletionListener` has been introduced so both implementations can keep up with finishing worker threads / scheduled jobs
- finally, a helper class named `ReaderUpdateServiceStarter`was introduced for consumers to be able to seamlessly call `startService`and let it handle which implementation to use, depending on the platform we're running on.

This pattern will be followed in other Services in the list in https://github.com/wordpress-mobile/WordPress-Android/pull/7506/#issuecomment-378949294, and we'll use the very handy `JobIntentService` for any other remaining `IntentServices` out there (specifically from that list, `PublicizeUpdateService`).

To test:
- on API26, run the app and place breakpoints in `ReaderUpdateJobService` to make sure it's executed.
- On previous Android versions, run the app and place breakpoints in `ReaderUpdateService` to make sure it's executed

cc @nbradbury

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateJobService.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                tasks.add(ReaderUpdateLogic.UpdateTask.values()[i]);
+            }
+            mReaderUpdateLogic.performTasks(tasks, params);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onStopJob(JobParameters params) {
+        AppLog.i(AppLog.T.READER, "reader service > stopped");
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateJobService.java]: Minor nit: can we change these to use "reader job service" to differentiate them from the existing logging in `ReaderUpdateService`?

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateJobService.java]: addressed in 8fad072

File: WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateJobService.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import org.wordpress.android.WordPress;
+import org.wordpress.android.util.AppLog;
+
+import java.util.EnumSet;
+
+import static org.wordpress.android.ui.reader.services.update.ReaderUpdateServiceStarter.ARG_UPDATE_TASKS;
+
+@TargetApi(21)
+public class ReaderUpdateJobService extends JobService implements ServiceCompletionListener {
+    /***
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateJobService.java]: We have this comment here, in `ReaderUpdateService`, and `ReaderUpdateLogic`. I think we can remove it from both services, and re-word it in `ReaderUpdateLogic` to better reflect the purpose of that class.

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateJobService.java]: addressed in 20793d9

**Comment by nbradbury** [comment]
Looks good - I'll merge when Travis is ready :shipit:

**Pull request merged**